types/sign.go

```diff
@@ -2,42 +2,30 @@ package types
 
 import (
 	"crypto/ecdsa"
-	"math/big"
+	"errors"
 	"strings"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
-// Sign the hashToSIgn
+var (
+	// ErrNonCanonicalSignature is returned when the signature is not canonical.
+	ErrNonCanonicalSignature = errors.New("received non-custodial signature")
```

Comment:
Non custodial or non canonical signature, or is it the same in fact?

Reply:
Fixed